circleci: failed tests: TestNodeRestart

The following test appears to have failed:

[#17206](https://circleci.com/gh/cockroachdb/cockroach/17206):

```
  cockroach: [start --ca-cert=/certs/ca.crt --cert=/certs/node.crt --key=/certs/node.key --host=roach1 --alsologtostderr=INFO --store=path=/data2.0 --join=roach0:26257 --log-dir=/logs/roach1 --logtostderr=false]
I160503 19:02:22.835375 acceptance/cluster/docker.go:328  unable to create container roach2: Error response from daemon: Conflict. The name "/roach2" is already in use by container f87fe142ce885d7ce87959e82b8d150f9fae7828c7cb932c9488e6c8e4351034. You have to remove (or rename) that container to be able to reuse that name.
I160503 19:02:22.836763 acceptance/cluster/localcluster.go:653  stopping
I160503 19:02:22.836893 acceptance/cluster/localcluster.go:551  event stream done, resetting...: net/http: request canceled
I160503 19:02:22.836904 acceptance/cluster/localcluster.go:574  events monitor exits
--- FAIL: TestNodeRestart (11.18s)
panic: error indicated existing container roach2, but none found:
error: Error response from daemon: Conflict. The name "/roach2" is already in use by container f87fe142ce885d7ce87959e82b8d150f9fae7828c7cb932c9488e6c8e4351034. You have to remove (or rename) that container to be able to reuse that name.
containers: [{ID:5b22ea23d1e7f567c557b47cc4982936dbd815cca0528c6df9f7daec7e452502 Names:[/roach1] Image:cockroachdb/builder:20160305-182433 ImageID:sha256:26823585ed7ee28fa547f4a91d16d6f26765bfcc3b49e8befcd6f67189ad4617 Command:/cockroach start --ca-cert=/certs/ca.crt --cert=/certs/node.crt --key=/certs/node.key --host=roach1 --alsologtostderr=INFO --store=path=/data2.0 --join=roach0:26257 --log-dir=/logs/roach1 --logtostderr=false Created:1462302132 Ports:[{IP:0.0.0.0 PrivatePort:26257 PublicPort:32798 Type:tcp} {IP:0.0.0.0 PrivatePort:8080 PublicPort:32799 Type:tcp}] SizeRw:0 SizeRootFs:0 Labels:map[Roach: Hostname:roach1] State: Status:Up 10 seconds HostConfig:{NetworkMode:598508a6052a2b62c0edfb335e45a2d4d7c9082323df7dfa041616e765bd74f0} NetworkSettings:0xc820118350 Mounts:[]} {ID:d6774258776d5645015184d11363f22481a2f8d5d6fc22582d8ce9bee5ee2a0a Names:[/roach0] Image:cockroachdb/builder:20160305-182433 ImageID:sha256:26823585ed7ee28fa547f4a91d16d6f26765bfcc3b49e8befcd6f67189ad4617 Command:/cockroach start --ca-cert=/certs/ca.crt --cert=/certs/node.crt --key=/certs/node.key --host=roach0 --alsologtostderr=INFO --store=path=/data1.0 --log-dir=/logs/roach0 --logtostderr=false Created:1462302132 Ports:[{IP:0.0.0.0 PrivatePort:26257 PublicPort:32796 Type:tcp} {IP:0.0.0.0 PrivatePort:8080 PublicPort:32797 Type:tcp}] SizeRw:0 SizeRootFs:0 Labels:map[Hostname:roach0 Roach:] State: Status:Up 10 seconds HostConfig:{NetworkMode:598508a6052a2b62c0edfb335e45a2d4d7c9082323df7dfa041616e765bd74f0} NetworkSettings:0xc820118360 Mounts:[]} {ID:f5d9495377174b6da5d8b1e25d92836386f99321db82584f80674cdd26785a29 Names:[/volumes] Image:cockroachdb/builder:20160305-182433 ImageID:sha256:26823585ed7ee28fa547f4a91d16d6f26765bfcc3b49e8befcd6f67189ad4617 Command:/bin/true Created:1462302131 Ports:[] SizeRw:0 SizeRootFs:0 Labels:map[] State: Status:Exited (0) 10 seconds ago HostConfig:{NetworkMode:598508a6052a2b62c0edfb335e45a2d4d7c9082323df7dfa041616e765bd74f0} NetworkSettings:0xc820118370 Mounts:[]} {ID:6d0f67b685745c47827877f6ad958e0b36a1fde92d142a578d300e8f158a8e45 Names:[/dreamy_ramanujan] Image:cockroachdb/builder:20160305-182433 ImageID:sha256:26823585ed7ee28fa547f4a91d16d6f26765bfcc3b49e8befcd6f67189ad4617 Command:make test 'TESTFLAGS=-v --verbosity=1 --vmodule=monitor=2,tracer=2' Created:1462301779 Ports:[] SizeRw:0 SizeRootFs:0 Labels:map[] State: Status:Exited (0) 2 minutes ago HostConfig:{NetworkMode:default} NetworkSettings:0xc820118380 Mounts:[]} {ID:fe432ad58414777580b2acb8fdb51c4a23b4b11c83b2cd14683ab3d350cef37a Names:[/nostalgic_euler] Image:cockroachdb/builder:20160305-182433 ImageID:sha256:26823585ed7ee28fa547f4a91d16d6f26765bfcc3b49e8befcd6f67189ad4617 Command:build/circle-deps.sh docker Created:1462301661 Ports:[] SizeRw:0 SizeRootFs:0 Labels:map[] State: Status:Exited (0) 6 minutes ago HostConfig:{NetworkMode:default} NetworkSettings:0xc820118390 Mounts:[]}] [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x6d94a4]

goroutine 648 [running]:
panic(0x1319560, 0xc820014140)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
testing.tRunner.func1(0xc82040a090)
	/usr/local/go/src/testing/testing.go:467 +0x192
panic(0x1319560, 0xc820014140)
	/usr/local/go/src/runtime/panic.go:426 +0x4e9
github.com/cockroachdb/cockroach/acceptance/cluster.(*LocalCluster).expectEvent(0xc8200c22a0, 0xc8204a0a20, 0xc82020c8e8, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/acceptance/cluster/localcluster.go:183 +0x274
github.com/cockroachdb/cockroach/acceptance/cluster.(*Container).Kill(0xc8204a0a20, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/acceptance/cluster/docker.go:187 +0x20e
github.com/cockroachdb/cockroach/acceptance/cluster.(*LocalCluster).stop(0xc8200c22a0)
	/go/src/github.com/cockroachdb/cockroach/acceptance/cluster/localcluster.go:664 +0x1a0
github.com/cockroachdb/cockroach/acceptance/cluster.(*LocalCluster).stopOnPanic(0xc8200c22a0)
	/go/src/github.com/cockroachdb/cockroach/acceptance/cluster/localcluster.go:236 +0x4d
panic(0x12cc860, 0xc8201ea2f0)
	/usr/local/go/src/runtime/panic.go:426 +0x4e9
github.com/cockroachdb/cockroach/acceptance/cluster.maybePanic(0x7f7af892f028, 0xc8201ea2f0)
	/go/src/github.com/cockroachdb/cockroach/acceptance/cluster/docker.go:166 +0x4b
github.com/cockroachdb/cockroach/acceptance/cluster.(*LocalCluster).createRoach(0xc8200c22a0, 0xc8203d3020, 0xc8204a0a20, 0xc8203eb840, 0x2, 0x2, 0xc820272540, 0xa, 0xc)
	/go/src/github.com/cockroachdb/cockroach/acceptance/cluster/localcluster.go:420 +0x792
github.com/cockroachdb/cockroach/acceptance/cluster.(*LocalCluster).startNode(0xc8200c22a0, 0xc8203d3020)
	/go/src/github.com/cockroachdb/cockroach/acceptance/cluster/localcluster.go:477 +0x920
github.com/cockroachdb/cockroach/acceptance/cluster.(*LocalCluster).Start(0xc8200c22a0)
	/go/src/github.com/cockroachdb/cockroach/acceptance/cluster/localcluster.go:598 +0x7a1
github.com/cockroachdb/cockroach/acceptance.StartCluster(0xc82040a090, 0xc8205d6470, 0x9, 0xc8205c0880, 0x1, 0x1, 0x12a05f200, 0x0, 0x0)
--
testing.tRunner(0xc82040a090, 0x1daa1a0)
	/usr/local/go/src/testing/testing.go:473 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:582 +0x892

goroutine 1 [chan receive]:
testing.RunTests(0x173fd40, 0x1daa140, 0x17, 0x17, 0xc82025bd01)
	/usr/local/go/src/testing/testing.go:583 +0x8d2
testing.(*M).Run(0xc82027af08, 0x173dfd0)
	/usr/local/go/src/testing/testing.go:515 +0x81
github.com/cockroachdb/cockroach/acceptance.TestMain(0xc82027af08)
	/go/src/github.com/cockroachdb/cockroach/acceptance/main_test.go:48 +0x3e
main.main()
	github.com/cockroachdb/cockroach/acceptance/_test/_testmain.go:96 +0x114

goroutine 17 [syscall, 2 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1998 +0x1

goroutine 20 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x2082200)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1011 +0x64
created by github.com/cockroachdb/cockroach/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:598 +0x8a

goroutine 22 [syscall, 2 minutes]:
os/signal.signal_recv(0x0)
	/usr/local/go/src/runtime/sigqueue.go:116 +0x132
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 23 [chan receive, 2 minutes]:
github.com/cockroachdb/cockroach/acceptance.TestMain.func1()
	/go/src/github.com/cockroachdb/cockroach/acceptance/main_test.go:39 +0xd8
created by github.com/cockroachdb/cockroach/acceptance.TestMain
	/go/src/github.com/cockroachdb/cockroach/acceptance/main_test.go:47 +0x30

goroutine 572 [chan receive, 1 minutes]:
database/sql.(*DB).connectionOpener(0xc820010790)
	/usr/local/go/src/database/sql/sql.go:727 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x33f

goroutine 29 [select, 2 minutes, locked to thread]:
runtime.gopark(0x1740580, 0xc820025728, 0x1466c50, 0x6, 0x18, 0x2)
	/usr/local/go/src/runtime/proc.go:262 +0x163
runtime.selectgoImpl(0xc820025728, 0x0, 0x18)
	/usr/local/go/src/runtime/select.go:392 +0xa67
runtime.selectgo(0xc820025728)
	/usr/local/go/src/runtime/select.go:215 +0x12
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal1_unix.go:279 +0x358
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1998 +0x1

goroutine 27 [IO wait, 2 minutes]:
net.runtime_pollWait(0x7f7af8900dc8, 0x72, 0xc82047c000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc8203e2c30, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8203e2c30, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8203e2bd0, 0xc82047c000, 0x1000, 0x1000, 0x0, 0x7f7af892f050, 0xc820014200)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc8201186b8, 0xc82047c000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc8203ff930)
	/usr/local/go/src/net/http/transport.go:1069 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:853 +0x10a6

goroutine 28 [select, 2 minutes]:
net/http.(*persistConn).writeLoop(0xc8203ff930)
	/usr/local/go/src/net/http/transport.go:1273 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:854 +0x10cb

goroutine 531 [chan receive, 1 minutes]:
database/sql.(*DB).connectionOpener(0xc8200ea630)
	/usr/local/go/src/database/sql/sql.go:727 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x33f

goroutine 142 [IO wait, 1 minutes]:
net.runtime_pollWait(0x7f7af8900c48, 0x72, 0xc8202df000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc8203e3170, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8203e3170, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8203e3110, 0xc8202df000, 0x1000, 0x1000, 0x0, 0x7f7af892f050, 0xc820014200)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc82002a248, 0xc8202df000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc820124d00)
	/usr/local/go/src/net/http/transport.go:1069 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:853 +0x10a6

goroutine 270 [IO wait, 1 minutes]:
net.runtime_pollWait(0x7f7af8900a08, 0x72, 0xc820205000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc8203e21b0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8203e21b0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8203e2150, 0xc820205000, 0x1000, 0x1000, 0x0, 0x7f7af892f050, 0xc820014200)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc820118100, 0xc820205000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc8203fe680)
	/usr/local/go/src/net/http/transport.go:1069 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:853 +0x10a6

goroutine 432 [select]:
net/http.(*persistConn).writeLoop(0xc820060820)
	/usr/local/go/src/net/http/transport.go:1273 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:854 +0x10cb

goroutine 143 [select, 1 minutes]:
net/http.(*persistConn).writeLoop(0xc820124d00)
	/usr/local/go/src/net/http/transport.go:1273 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:854 +0x10cb

goroutine 506 [chan receive, 1 minutes]:
database/sql.(*DB).connectionOpener(0xc8203b7550)
	/usr/local/go/src/database/sql/sql.go:727 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x33f

goroutine 431 [IO wait]:
net.runtime_pollWait(0x7f7af8900ac8, 0x72, 0xc82039a000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc820552300, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820552300, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8205522a0, 0xc82039a000, 0x1000, 0x1000, 0x0, 0x7f7af892f050, 0xc820014200)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc820118278, 0xc82039a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc820060820)
	/usr/local/go/src/net/http/transport.go:1069 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:853 +0x10a6

goroutine 271 [select, 1 minutes]:
net/http.(*persistConn).writeLoop(0xc8203fe680)
	/usr/local/go/src/net/http/transport.go:1273 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:854 +0x10cb

goroutine 698 [IO wait]:
net.runtime_pollWait(0x7f7af89001c8, 0x72, 0xc820234000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc8201d4140, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201d4140, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8201d40e0, 0xc820234000, 0x1000, 0x1000, 0x0, 0x7f7af892f050, 0xc820014200)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc820118340, 0xc820234000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc820060b60)
	/usr/local/go/src/net/http/transport.go:1069 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:853 +0x10a6

goroutine 349 [IO wait, 1 minutes]:
net.runtime_pollWait(0x7f7af8900b88, 0x72, 0xc8202c9000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc8205521b0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8205521b0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820552150, 0xc8202c9000, 0x1000, 0x1000, 0x0, 0x7f7af892f050, 0xc820014200)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc820118238, 0xc8202c9000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc820060270)
	/usr/local/go/src/net/http/transport.go:1069 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:853 +0x10a6

goroutine 350 [select, 1 minutes]:
net/http.(*persistConn).writeLoop(0xc820060270)
	/usr/local/go/src/net/http/transport.go:1273 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:854 +0x10cb

goroutine 502 [chan receive, 1 minutes]:
database/sql.(*DB).connectionOpener(0xc8203b7340)
	/usr/local/go/src/database/sql/sql.go:727 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x33f

goroutine 498 [chan receive, 1 minutes]:
database/sql.(*DB).connectionOpener(0xc8203b6f20)
	/usr/local/go/src/database/sql/sql.go:727 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x33f

goroutine 699 [select]:
net/http.(*persistConn).writeLoop(0xc820060b60)
	/usr/local/go/src/net/http/transport.go:1273 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:854 +0x10cb

goroutine 494 [chan receive, 1 minutes]:
database/sql.(*DB).connectionOpener(0xc8203b68f0)
	/usr/local/go/src/database/sql/sql.go:727 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x33f

goroutine 702 [runnable]:
github.com/docker/engine-api/client/transport/cancellable.Do.func3(0x7f7af8900f30, 0xc8205daab0, 0xc8205d71e0, 0xc8202dd440)
	/go/src/github.com/docker/engine-api/client/transport/cancellable/cancellable.go:77
created by github.com/docker/engine-api/client/transport/cancellable.Do
	/go/src/github.com/docker/engine-api/client/transport/cancellable/cancellable.go:84 +0x336

goroutine 529 [chan receive, 1 minutes]:
database/sql.(*DB).connectionOpener(0xc820011b80)
	/usr/local/go/src/database/sql/sql.go:727 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x33f
ok github.com/cockroachdb/cockroach/acceptance 1337s

```
Please assign, take a look and update the issue accordingly.

--- a/acceptance/cluster/docker.go
+++ b/acceptance/cluster/docker.go
@@ -350,9 +350,14 @@ func (cli resilientDockerClient) ContainerCreate(
 				return cli.ContainerCreate(ctx, config, hostConfig, networkingConfig, containerName)
 			}
 		}
-		return response, fmt.Errorf("error indicated existing container %s, "+
+		log.Warningf("error indicated existing container %s, "+
 			"but none found:\nerror: %s\ncontainers: %+v",
 			containerName, err, containers)
+		// We likely raced with a previous (late) removal of the container.
+		// Return a timeout so a higher level can retry and hopefully
+		// succeed (or get stuck in an infinite loop, at which point at
+		// least we'll have gathered an additional bit of information).
+		return response, context.DeadlineExceeded
 	}
 	return response, err
 }

